content/docs/command-reference/exp/run.md

```diff
@@ -134,6 +141,17 @@ CPU cores).
 - `-j <number>`, `--jobs <number>` - run this `number` of queued experiments in
   parallel. Only applicable when used in conjunction with `--run-all`.
 
+- `-r <commit>`, `--rev <commit>` - continue an experiment from a specific
+  checkpoint name or hash (`commit`). Required for explicitly resuming
+  checkpoints in queued or temp dir runs.
```

Comment:
> Required for explicitly resuming checkpoints in queued or temp dir runs.

Got that from the CLI @pmrowla but TBH I don't know what it means 😬. Could you clarify? Thanks

Reply:
This is out of date, the current CLI message is 
```
  -r <experiment_rev>, --rev <experiment_rev>
                        Continue the specified checkpoint experiment. (Only
                        required for explicitly resuming checkpoints in queued
                        or temp dir runs.)
```

The old message was from when the default behavior for `exp run` was to always run from scratch (and reset checkpoints). To resume from an existing checkpoint, you would need to specify the revision to resume from.

The current behavior is that `exp run` with no args will resume the last run checkpoint by default (in the workspace), and `--reset` must be used to explicitly restart from scratch.

The `-r/--rev` argument is still required when the user wants to resume from a previous checkpoint  using `--queue` or `--temp`, since it is not obvious what point DVC should resume from in that situation.

Reply:
OK got it (updated).

I'm still seeing ` (Only required for explicitly resuming checkpoints in queued or temp dir runs.)` in DVC `2.0.0a3` (latest `master`) though.

> The -r/--rev argument is still required when the user wants to resume from a previous checkpoint using --queue or --temp, since it is not obvious what point DVC should resume from in that situation.

This part is still confusing to me. Are you saying that if the run is interrupted while there are still queued exps not run, and the run exps registered some checkpoints, DVC doesn't know wha to do? What options do you have as user to send in `rev` in that case? Thanks

Reply:
It means that if you want each specific queued run or temp directory run to be a "resume" run, you have to specify what checkpoint commit to resume from, otherwise any queued or temp directory run will always be a "reset" run starting from HEAD by default.

It's independent of whatever else is in the queue. The explicit parameter is required because when using queue+run-all or temp dir runs, the workspace is not modified (so it always just contains `HEAD`).

Let's say I queue 5 different sets of checkpoint experiments (with unique parameters for each set of checkpoints), and then use `--run-all`. Now If I just said `dvc exp run`, DVC has no way of knowing which of the 5 experiments which were just executed should be resumed.

So if I want to continue the second and third sets of checkpoints, I have to explicitly do

```
dvc exp run --queue -r <last_checkpoint_in_2>
dvc exp run --queue -r <last_checkpoint_in_3>
```

Now my next `--run-all` will continue generating two checkpoint runs, one which is resuming from `1`, and one which is resuming from `3`

Reply:
Oops this comment got mixed with my `exp gc` https://github.com/iterative/dvc.org/pull/2243#pullrequestreview-601464535... Hope it's not too confusing!

Reply:
> means that if you want each specific queued run or temp directory run to be a "resume" run, you have to specify what checkpoint commit to resume from

Here I still wasn't getting how different queued checkpoint exps can share a single starting point if they all have their own checkpoints (different revs)... But I was assuming we're talking about `exp run --run-all` in every case.

> when using queue+run-all or temp dir runs, the workspace is not modified
> If I just tried to do dvc exp run, DVC has no way of knowing which...

This is much clearer, thanks! Updated in docs.

My only remaining Q on this is how to resume *multiple* checkpoint exps. Will `exp run --run-all` continue each one from their last checkpoint? Can you combine `--run-all` and `--rev`? Or perhaps that case is not supported

Reply:
`--run-all` just runs whatever is in your queue. If you want to resume multiple checkpoint exps, then you just have to queue them.

So do `dvc exp run --rev <something to continue> --queue` multiple times, with different exps in `<something to continue>`

Reply:
Maybe the confusion is about the `continue/resume` wording. Internally, the way that `--rev` works is that DVC just runs an experiment starting from `--rev` instead of HEAD. And if `--rev` is the tip of an existing experiment branch, new checkpoint commits will be appended to that same branch instead of creating a new experiment branch name.

Reply:
Aaaah I see. You can queue each one with a given `--rev` dah. Makes sense, good to know (I don't think I'll mention it in docs for now).